pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -272,6 +285,51 @@ func (m *AddOnManager) verifyRequiredVariablesInContext(context *command.Executi
 	return nil
 }
 
+func (m *AddOnManager) verifyRequiredOpenshiftVersion(context *command.ExecutionContext, meta addon.AddOnMeta) error {
+	dockerCommander := context.GetDockerCommander()
+	versionInfo, err := dockerCommander.Exec(" ", constants.OpenshiftContainerName, "openshift", "version")
+	if err != nil {
+		return err
+	}
+	openShiftVersionAlongWithCommitSha := strings.Split(strings.Split(versionInfo, "\n")[0], " ")[1]
+	openShiftVersion := strings.TrimSpace(strings.Split(openShiftVersionAlongWithCommitSha, "+")[0])
+	requiredOpenshiftVersion := strings.TrimSpace(meta.OpenShiftVersion())
+	if requiredOpenshiftVersion != "" {
+		versionErr := errors.New(fmt.Sprintf("Add-on only support OpenShift version as <major>.<minor>.<patch> (ex. v3.6.0) "+
+			"Provided %s\n", requiredOpenshiftVersion))
+		if strings.HasPrefix(requiredOpenshiftVersion, ">=") {
+			if !checkOpenShiftVersionSemantic(strings.TrimPrefix(requiredOpenshiftVersion, ">=")) {
+				return versionErr
+			}
+			// This will work for both upstream and downstream.
+			if util.VersionOrdinal(openShiftVersion) < util.VersionOrdinal(strings.TrimPrefix(requiredOpenshiftVersion, ">=")) {
+				return errors.New(fmt.Sprintf("Addon does not support OpenShift version %s. "+
+					"You need to use a version %s\n", openShiftVersion, requiredOpenshiftVersion))
+			}
+			return nil
+		}
+		if strings.HasPrefix(requiredOpenshiftVersion, ">") {
+			if !checkOpenShiftVersionSemantic(strings.TrimPrefix(requiredOpenshiftVersion, ">")) {
+				return versionErr
+			}
+			// This will work for both upstream and downstream.
+			if util.VersionOrdinal(openShiftVersion) <= util.VersionOrdinal(strings.TrimPrefix(requiredOpenshiftVersion, ">")) {
+				return errors.New(fmt.Sprintf("Addon does not support OpenShift version %s. "+
+					"You need to use a version %s\n", openShiftVersion, requiredOpenshiftVersion))
+			}
+			return nil
+		}
+		if !checkOpenShiftVersionSemantic(strings.TrimPrefix(requiredOpenshiftVersion, ">")) {
+			return versionErr
+		}
+		if openShiftVersion != requiredOpenshiftVersion {
+			return errors.New(fmt.Sprintf("Addon does not support OpenShift version %s. "+
+				"You need to use a version %s\n", openShiftVersion, requiredOpenshiftVersion))
+		}
+	}
```

Comment:
Modified the add-on to use `OpenShift-Version: >v3.6.0`
```
$ minishift addon apply admin-user
-- Applying addon 'admin-user':Error applying the add-on: Addon does not support OpenShift version v3.6.0. You need to use a version >v3.6.0
```
Modified the add-on to use `OpenShift-Version: <v3.6.0`
```
[lmohanty@LalatenduM-laptop minishift]$ minishift addon apply admin-user
-- Applying addon 'admin-user':Error applying the add-on: Add-on only support OpenShift version as <major>.<minor>.<patch> (ex. v3.6.0) Provided <v3.6.0
```
@praveenkumar can we please simplify the error and make it consistent.  For example the last error should be `Addon does not support OpenShift version v3.6.0. You need to use a version <v3.6.0.  Add-on only support OpenShift version as <major>.<minor>.<patch> (ex. v3.6.0)`  or something better.